scripts/signing/pldm_fw_package/pldm_update_pkg_gen.py

```diff
@@ -0,0 +1,291 @@
+    "pvt" : "PVT",
+    "mp" : "MP",
+}
+
+DBG_EN = True # Enable if there's no Sign EXE file
+
+# import common library
+comm_msg = Common_msg()
+msg_hdr_print = comm_msg.msg_hdr_print
+comm_file = Common_file()
+is_file_exist = comm_file.is_file_exist
+resource_path = comm_file.resource_path
+get_md5_str_from_file = comm_file.get_md5_str_from_file_list
+comm_sys = System_ctrl()
+platform_os = comm_sys.os_name.lower()
+
+# EXE file
+EXE_WIN_FILE = resource_path("pldm_fwup_pkg_creator.exe")
+EXE_LINUX_FILE = resource_path("./pldm_fwup_pkg_creator")
+
+if DBG_EN == True:
+    command_prefix = resource_path("pldm_fwup_pkg_creator.py")
+else:
+    if platform_os == "windows":
+        command_prefix = EXE_WIN_FILE
+        if not is_file_exist(EXE_WIN_FILE):
+            print("Can't find exe file " + EXE_WIN_FILE)
+            sys.exit(1)
+    elif platform_os == "linux":
+        command_prefix = EXE_LINUX_FILE
+        if not is_file_exist(EXE_LINUX_FILE):
+            print("Can't find exe file " + EXE_LINUX_FILE)
+            sys.exit(1)
+    else:
+        print("Current os " + platform_os + " is not supported!")
+        sys.exit(1)
+
+def APP_HELP():
+    msg_hdr_print("n", "--------------------------------------------------------------------", "\n")
+    msg_hdr_print("n", "HELP:")
+    msg_hdr_print("n", "-p      Platform select.")
+    msg_hdr_print("n", "          [gt] GrandTeton")
+    msg_hdr_print("n", "-b      Board select.")
+    msg_hdr_print("n", "          [swb] SwitchBoard")
+    msg_hdr_print("n", "-s      Stage select.")
+    msg_hdr_print("n", "          [poc/evt/dvt/pvt/mp]")
+    msg_hdr_print("n", "-c      Component id select.")
+    msg_hdr_print("n", "          Please follow spec, could be list(-c 1 2).")
+    msg_hdr_print("n", "-v      Component version string select.")
+    msg_hdr_print("n", '          Please follow spec, could be list(-v "ver1" "ver2").')
+    msg_hdr_print("n", "-i      Component image select.")
+    msg_hdr_print("n", "          Please follow spec, could be list(-i ./img1 ./img2).")
+    msg_hdr_print("n", "")
+    msg_hdr_print("n", "--------------------------------------------------------------------")
+
+def APP_HEADER():
+    msg_hdr_print("n", "========================================================")
+    msg_hdr_print("n", "* APP name:    "+APP_NAME)
+    msg_hdr_print("n", "* APP auth:    "+APP_AUTH)
+    msg_hdr_print("n", "* APP version: "+APP_RELEASE_VER)
+    msg_hdr_print("n", "* APP date:    "+APP_RELEASE_DATE)
+    msg_hdr_print("n", "* NOTE: This APP is based on pldm_fwup_pkg_creator.py")
+    msg_hdr_print("n", "========================================================")
+
+def PLAT_CheckVRChecksum(str, byte_num):
+    if len(str) < byte_num*2:
+        return False
+
+    focus_str = str[0:byte_num*2]
+    print(focus_str)
+    for c in focus_str:
+        try:
+            int(c, 16)
+        except:
+            return False
+
+    return True
+class APP_ARG():
```

Comment:
Why not use argparse? 
https://docs.python.org/3/library/argparse.html
Its builtin to standard python installs and would solve a lot of this code here for you.
```
import argparse

parser = argparse.ArgumentParser()
parser.add_argument("-p", "--platform", action="store", dest="platform_select", required=True, help="Platform select")
parser.add_argument("-b", "--board", required=True,  action="store", dest="board_select", help="Board select")
<etc>
args = parser.parse_args()
Use args.platform_select, args.board_select 
Help will be autopopulated so you dont need APP_HELP as well!

Reply:
Hi @amithash ,

Got it thanks, please have a look at new commit.